test/libsolidity/SMTCheckerTest.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/libsolidity/SMTCheckerTest.h>
+#include <test/Options.h>
+
+#include <libsolidity/formal/ModelChecker.h>
+
+using namespace langutil;
+using namespace dev::solidity;
+using namespace dev::solidity::test;
+using namespace dev;
+using namespace std;
+
+SMTCheckerTest::SMTCheckerTest(string const& _filename, langutil::EVMVersion _evmVersion): SyntaxTest(_filename, _evmVersion)
+{
+	if (m_settings.count("SMTSolvers"))
+	{
+		auto const& choice = m_settings.at("SMTSolvers");
+		if (choice == "any")
+			m_enabledSolvers = smt::SMTSolverChoice::All();
+		else if (choice == "z3")
+			m_enabledSolvers = smt::SMTSolverChoice::Z3();
+		else if (choice == "cvc4")
+			m_enabledSolvers = smt::SMTSolverChoice::CVC4();
+		else if (choice == "none")
+			m_enabledSolvers = smt::SMTSolverChoice::None();
+		else
+			BOOST_THROW_EXCEPTION(runtime_error("Invalid SMT solver choice."));
```

Comment:
Could use uniform initialization: `runtime_error{"Invalid SMT solver choice."}`

Reply:
Instead of BOOST...?

Reply:
Otherwise, what would be the reason for the change?